notte/llms/service.py

```diff
@@ -1,42 +1,26 @@
-import os
 from pathlib import Path
-from typing import Any, ClassVar, final
+from typing import Any, final
 
 from litellm import ModelResponse
+from llamux import Router
 
-from notte.llms.engine import LLMEngine
 from notte.llms.prompt import PromptLibrary
 
 PROMPT_DIR = Path(__file__).parent.parent / "llms" / "prompts"
-
-
-class ModelRouter:
-    NOTTE_BASE_MODEL: ClassVar[str] = "NOTTE_BASE_MODEL"
-
-    def get(self) -> str:
-        model = os.getenv(self.NOTTE_BASE_MODEL)
-        if model is None:
-            return "groq/llama-3.3-70b-versatile"
-        return model
-
-    @staticmethod
-    def set(model: str):
-        os.environ[ModelRouter.NOTTE_BASE_MODEL] = model
+ENDPOINT_DIR = Path(__file__).parent.parent / "llms" / "config" / "endpoints.csv"
 
 
 @final
 class LLMService:
 
     def __init__(self):
-        self.llm = LLMEngine()
         self.lib = PromptLibrary(str(PROMPT_DIR))
-        self.router = ModelRouter()
+        self.router = Router.from_csv(str(ENDPOINT_DIR))
```

Comment:
Should the router have a completion method ? I think it would be better to return the model to try next (its a router not a LLM service). In that way the routing can be optional in the notte library (users can chose to do that if they want. Maybe they want to stick with a single API_KEY